### ISSUE TITLE
YoloV3 prediction accuracy

### ISSUE THREAD

**Issue by HidayathullaShaik** [AlexeyAB/darknet#2086]
Hi,
I am trying to implement YoloV3 for weapon detection in video surveillance system.
I have trained for Gun, Knife Classes and the system is detecting decently on good resolution images but not on low quality images from video surviellance footages.

My machine hardware specification used for YoloV3 Training:
RAM: 16 GB
Processor: i7-8750H CPU @2.20 GHz x 12
Graphics: GeForce GTX 1060
CUDA: 10.0.130
CUDA Cores: 1280
OS: Ubuntu 18.04.1 LTS

I have already implemented the suggestions at: https://github.com/AlexeyAB/darknet#how-to-improve-object-detection

Please let me know if there is any thing that I could do to improve the prediction of video surveillance systems.

Thanks,
Hidayath

**Comment by AlexeyAB** [comment]
@HidayathullaShaik Hi,

> the system is detecting decently on good resolution images but not on low quality images from video surviellance footages.

You should train it on low quality images from video surviellance footages.

Also check this rule: https://github.com/AlexeyAB/darknet#how-to-improve-object-detection

> General rule - your training dataset should include such a set of relative sizes of objects that you want to detect:
> 
> `train_network_width * train_obj_width / train_image_width ~= detection_network_width * detection_obj_width / detection_image_width`

**Comment by nguyendu392** [comment]
Hello @AlexeyAB, 
I'm using the pre-trained yolov3.weights to detect the person. I edited `valid = path_of_my_test` in coco.data. 
But when I run `./darknet detector map cfg/coco.data yolov3.cfg yolov3.weights `, it calculates mAP all 80 classes. 
I want to calculate mAP only class person. Could you tell me how to do what I want?

**Comment by HidayathullaShaik** [comment]
> @HidayathullaShaik Hi,
> 
> > the system is detecting decently on good resolution images but not on low quality images from video surviellance footages.
> 
> You should train it on low quality images from video surviellance footages.
> 
> Also check this rule: https://github.com/AlexeyAB/darknet#how-to-improve-object-detection
> 
> > General rule - your training dataset should include such a set of relative sizes of objects that you want to detect:
> > `train_network_width * train_obj_width / train_image_width ~= detection_network_width * detection_obj_width / detection_image_width`

Hi Alexey,
I am not clear on the above mentioned terms in the rule.
I understand that "train_network_width" is the width configured in the CFG file used for training.
Similarly, "detection_network_width" is the width configured in CFG file used for detection/prediction. 
What are train_obj_width, train_image_width, detection_obj_width, detection_image_width.

Questions:
1. Could you please help me in understanding the widths mentioned in the general rule.
2. How do I ensure this rule across all the images in my dataset. Do I have to calculate it manually or Is there any command to check this?

Thanks,
Hidayath

**Comment by AlexeyAB** [comment]
@HidayathullaShaik 

> What are train_obj_width, train_image_width, detection_obj_width, detection_image_width.

It means that for each of image & object from Test dataset there must be at least 1 image in the Training dataset that satisfies this condition: 

`width of object from Training dataset` / `width of image from Training dataset` ~= 
`width of object from Test dataset` / `width of image from Test  dataset`

----

I.e. for each of object from Test dataset there must be at least 1 object in the Training dataset with the same relative size:

`object width in percent from Training dataset` ~= `object width in percent from Test dataset` 


That is, if in the training set there were only objects that occupied 80-90% of the image, then the trained network will not be able to detect objects that occupy 1-10% of the image.

**Comment by HidayathullaShaik** [comment]
Thanks a lot Alexey. I will ensure this rule during training.

Thanks,
Hidayath
